src/types/IUiaTraceable.h

```diff
@@ -0,0 +1,48 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- IUiaTraceable.hpp
+
+Abstract:
+- This module is used for assigning and retrieving IDs to Uia objects
+
+Author(s):
+- Carlos Zamora (cazamor)     Apr 2020
+--*/
+
+#pragma once
+
+namespace Microsoft::Console::Types
+{
+    typedef unsigned long long IdType;
+    constexpr IdType InvalidId = 0;
+
+    class IUiaTraceable
+    {
+    public:
+        const IdType GetId() const noexcept
+        {
+            return _id;
+        }
+
+        bool AssignId(IdType id) noexcept

```

Comment:
And maybe add a doc comment here too, because this looks like it's designed as something that will get called multiple times for the same object, and only store the ID the first time.